### PR TITLE
.*: remove delta mode of execution summary

### PR DESCRIPTION
**Pull request by SeaRise**
### What problem does this PR solve?

Issue Number: ref #6233

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/DataStreams/TiRemoteBlockInputStream.h [hunk 1]
```diff
@@ -67,65 +67,25 @@ class TiRemoteBlockInputStream : public IProfilingBlockInputStream
...
+        if (unlikely(resp.execution_summaries_size() == 0))
             return;
 
-        if (!execution_summaries_inited[index].load())
-        {
-            initRemoteExecutionSummaries(resp, index);
-            return;
-        }
-        auto & execution_summaries_map = execution_summaries[index];
+        // There will only be one summary packet for one index.
```
Comment (windtalker) [inline dbms/src/DataStreams/TiRemoteBlockInputStream.h:75]: This is not true for remote read :(, maybe we should seperate the execution summary part for `CoprocessorReader` and `ExchangeReceiver`.

Comment (SeaRise) [reply dbms/src/DataStreams/TiRemoteBlockInputStream.h:75]: O.....yes...CoprocessReader will send multiple cop tasks....

File: dbms/src/DataStreams/TiRemoteBlockInputStream.h [hunk 2]
```diff
@@ -67,65 +67,25 @@ class TiRemoteBlockInputStream : public IProfilingBlockInputStream
...
+        RUNTIME_CHECK(!execution_summaries_inited[index].load());
         for (const auto & execution_summary : resp.execution_summaries())
         {
-            if (execution_summary.has_executor_id())
+            if (likely(execution_summary.has_executor_id()))
             {
-                const auto & executor_id = execution_summary.executor_id();
-                if (unlikely(execution_summaries_map.find(executor_id) == execution_summaries_map.end()))
```
Comment (ywqzzy) [inline dbms/src/DataStreams/TiRemoteBlockInputStream.h:102]: Why remove line 102-106？

Comment (SeaRise) [reply dbms/src/DataStreams/TiRemoteBlockInputStream.h:102]: My mistake, I'll revert it...

**Comment by windtalker** [comment]
I'm not sure if this change compatible with `BatchCorpcessor`, please help confirm this.

**Comment by SeaRise** [comment]
> I'm not sure if this change compatible with `BatchCorpcessor`, please help confirm this.

looks ok

- mpp
+------------------------------------------------+------------+---------+--------------+---------------+---------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------+------+
| id                                             | estRows    | actRows | task         | access object | execution info                                                                                          | operator info                                                                                                                                                           | memory | disk |
+------------------------------------------------+------------+---------+--------------+---------------+---------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------+------+
| TableReader_82                                 | 5.17       | 5       | root         |               | time:1.34s, loops:2, cop_task: {num: 2, max: 0s, min: 0s, avg: 0s, p95: 0s, copr_cache_hit_ratio: 0.00} | data:ExchangeSender_81                                                                                                                                                  | N/A    | N/A  |
| └─ExchangeSender_81                            | 5.17       | 5       | mpp[tiflash] |               | tiflash_task:{time:1.33s, loops:1, threads:20}                                                          | ExchangeType: PassThrough                                                                                                                                               | N/A    | N/A  |
|   └─Projection_76                              | 5.17       | 5       | mpp[tiflash] |               | tiflash_task:{time:1.32s, loops:1, threads:20}                                                          | Column#8, Column#9                                                                                                                                                      | N/A    | N/A  |
|     └─HashAgg_74                               | 5.17       | 5       | mpp[tiflash] |               | tiflash_task:{time:1.32s, loops:1, threads:20}                                                          | group by:Column#36, funcs:sum(Column#34)->Column#8, funcs:sum(Column#35)->Column#9                                                                                      | N/A    | N/A  |
|       └─Projection_84                          | 5.17       | 5       | mpp[tiflash] |               | tiflash_task:{time:1.32s, loops:1, threads:20}                                                          | cast(test.test.col1, decimal(5,0) UNSIGNED BINARY)->Column#34, cast(test.test.col3, decimal(20,0) BINARY)->Column#35, Column#7                                          | N/A    | N/A  |
|         └─ExchangeReceiver_54                  | 5.17       | 5       | mpp[tiflash] |               | tiflash_task:{time:1.31s, loops:1, threads:20}                                                          |                                                                                                                                                                         | N/A    | N/A  |
|           └─ExchangeSender_53                  | 5.17       | 5       | mpp[tiflash] |               | tiflash_task:{time:1.31s, loops:1, threads:20}                                                          | ExchangeType: HashPartition, Hash Cols: [name: Column#7, collate: binary]                                                                                               | N/A    | N/A  |
|             └─Projection_49                    | 5.17       | 5       | mpp[tiflash] |               | tiflash_task:{time:1.31s, loops:1, threads:20}                                                          | Column#7, test.test.col1, test.test.col3                                                                                                                                | N/A    | N/A  |
|               └─HashAgg_50                     | 5.17       | 5       | mpp[tiflash] |               | tiflash_task:{time:1.31s, loops:1, threads:20}                                                          | group by:test.test.col1, test.test.col3, funcs:sum(Column#19)->Column#7, funcs:firstrow(test.test.col1)->test.test.col1, funcs:firstrow(test.test.col3)->test.test.col3 | N/A    | N/A  |
|                 └─ExchangeReceiver_52          | 5.17       | 5       | mpp[tiflash] |               | tiflash_task:{time:1.3s, loops:1, threads:20}                                                           |                                                                                                                                                                         | N/A    | N/A  |
|                   └─ExchangeSender_51          | 5.17       | 5       | mpp[tiflash] |               | tiflash_task:{time:1.31s, loops:1, threads:2}                                                           | ExchangeType: HashPartition, Hash Cols: [name: test.test.col1, collate: binary], [name: test.test.col3, collate: binary]                                                | N/A    | N/A  |
|                     └─HashAgg_45               | 5.17       | 5       | mpp[tiflash] |               | tiflash_task:{time:1.3s, loops:1, threads:2}                                                            | group by:Column#32, Column#33, funcs:sum(Column#31)->Column#19                                                                                                          | N/A    | N/A  |
|                       └─Projection_83          | 2322880.00 | 2322880 | mpp[tiflash] |               | tiflash_task:{time:848.4ms, loops:37, threads:2}                                                        | cast(test.test.col2, double BINARY)->Column#31, test.test.col1, test.test.col3                                                                                          | N/A    | N/A  |
|                         └─TableFullScan_34     | 2322880.00 | 2322880 | mpp[tiflash] | table:test    | tiflash_task:{time:181.4ms, loops:37, threads:2}                                                        | keep order:false                                                                                                                                                        | N/A    | N/A  |
+------------------------------------------------+------------+---------+--------------+---------------+---------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------+------+
- batchCop
+----------------------------------+------------+---------+-------------------+---------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+------+
| id                               | estRows    | actRows | task              | access object | execution info                                                                                                                                                                                                                                                                                                                                                        | operator info                                                                                                                                                           | memory    | disk |
+----------------------------------+------------+---------+-------------------+---------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+------+
| HashAgg_9                        | 5.17       | 5       | root              |               | time:1.16s, loops:4, partial_worker:{wall_time:1.158933148s, concurrency:5, task_num:4, tot_wait:5.794392244s, tot_exec:73.671µs, tot_time:5.794488045s, max:1.158902885s, p95:1.158902885s}, final_worker:{wall_time:1.15898255s, concurrency:5, task_num:4, tot_wait:5.794554389s, tot_exec:139.034µs, tot_time:5.794698452s, max:1.158953888s, p95:1.158953888s}   | group by:Column#18, funcs:sum(Column#16)->Column#8, funcs:sum(Column#17)->Column#9                                                                                      | 37.9 KB   | N/A  |
| └─Projection_19                  | 5.17       | 5       | root              |               | time:1.16s, loops:5, Concurrency:OFF                                                                                                                                                                                                                                                                                                                                  | cast(test.test.col1, decimal(5,0) UNSIGNED BINARY)->Column#16, cast(test.test.col3, decimal(20,0) BINARY)->Column#17, Column#7                                          | 1.11 KB   | N/A  |
|   └─HashAgg_14                   | 5.17       | 5       | root              |               | time:1.16s, loops:5, partial_worker:{wall_time:1.158761518s, concurrency:5, task_num:1, tot_wait:5.793408843s, tot_exec:28.892µs, tot_time:5.793454968s, max:1.158731855s, p95:1.158731855s}, final_worker:{wall_time:1.158790169s, concurrency:5, task_num:4, tot_wait:5.793568277s, tot_exec:248.29µs, tot_time:5.79382232s, max:1.158774266s, p95:1.158774266s}    | group by:test.test.col1, test.test.col3, funcs:sum(Column#10)->Column#7, funcs:firstrow(test.test.col1)->test.test.col1, funcs:firstrow(test.test.col3)->test.test.col3 | 17.3 KB   | N/A  |
|     └─TableReader_15             | 5.17       | 5       | root              |               | time:1.16s, loops:2, cop_task: {num: 2, max: 0s, min: 0s, avg: 0s, p95: 0s, copr_cache_hit_ratio: 0.00, distsql_concurrency: 15}                                                                                                                                                                                                                                      | data:HashAgg_10                                                                                                                                                         | 152 Bytes | N/A  |
|       └─HashAgg_10               | 5.17       | 5       | batchCop[tiflash] |               | tiflash_task:{time:1.15s, loops:1, threads:2}                                                                                                                                                                                                                                                                                                                         | group by:Column#14, Column#15, funcs:sum(Column#13)->Column#10                                                                                                          | N/A       | N/A  |
|         └─Projection_18          | 2322880.00 | 2322880 | batchCop[tiflash] |               | tiflash_task:{time:743.9ms, loops:37, threads:2}                                                                                                                                                                                                                                                                                                                      | cast(test.test.col2, double BINARY)->Column#13, test.test.col1, test.test.col3                                                                                          | N/A       | N/A  |
|           └─TableFullScan_13     | 2322880.00 | 2322880 | batchCop[tiflash] | table:test    | tiflash_task:{time:77.9ms, loops:37, threads:2}                                                                                                                                                                                                                                                                                                                       | keep order:false                                                                                                                                                        | N/A       | N/A  |
+----------------------------------+------------+---------+-------------------+---------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+------+
- cop
+----------------------------------+------------+---------+--------------+---------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+------+
| id                               | estRows    | actRows | task         | access object | execution info                                                                                                                                                                                                                                                                                                                                                       | operator info                                                                                                                                                           | memory    | disk |
+----------------------------------+------------+---------+--------------+---------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+------+
| HashAgg_9                        | 5.17       | 5       | root         |               | time:1.23s, loops:4, partial_worker:{wall_time:1.225490746s, concurrency:5, task_num:4, tot_wait:6.127030474s, tot_exec:31.683µs, tot_time:6.127071437s, max:1.225452897s, p95:1.225452897s}, final_worker:{wall_time:0s, concurrency:5, task_num:4, tot_wait:6.12737545s, tot_exec:80.303µs, tot_time:6.127458778s, max:1.225499416s, p95:1.225499416s}             | group by:Column#18, funcs:sum(Column#16)->Column#8, funcs:sum(Column#17)->Column#9                                                                                      | 37.9 KB   | N/A  |
| └─Projection_19                  | 5.17       | 5       | root         |               | time:1.23s, loops:5, Concurrency:OFF                                                                                                                                                                                                                                                                                                                                 | cast(test.test.col1, decimal(5,0) UNSIGNED BINARY)->Column#16, cast(test.test.col3, decimal(20,0) BINARY)->Column#17, Column#7                                          | 1.11 KB   | N/A  |
|   └─HashAgg_14                   | 5.17       | 5       | root         |               | time:1.23s, loops:5, partial_worker:{wall_time:1.225280629s, concurrency:5, task_num:1, tot_wait:6.12612229s, tot_exec:37.253µs, tot_time:6.126172144s, max:1.225264616s, p95:1.225264616s}, final_worker:{wall_time:1.225299391s, concurrency:5, task_num:4, tot_wait:6.126260169s, tot_exec:143.743µs, tot_time:6.12640806s, max:1.225288809s, p95:1.225288809s}   | group by:test.test.col1, test.test.col3, funcs:sum(Column#10)->Column#7, funcs:firstrow(test.test.col1)->test.test.col1, funcs:firstrow(test.test.col3)->test.test.col3 | 17.3 KB   | N/A  |
|     └─TableReader_15             | 5.17       | 10      | root         |               | time:1.23s, loops:2, cop_task: {num: 2, max: 1.23s, min: 1.15s, avg: 1.19s, p95: 1.23s, rpc_num: 2, rpc_time: 2.38s, copr_cache_hit_ratio: 0.00, distsql_concurrency: 15}                                                                                                                                                                                            | data:HashAgg_10                                                                                                                                                         | 638 Bytes | N/A  |
|       └─HashAgg_10               | 5.17       | 10      | cop[tiflash] |               | tiflash_task:{proc max:1.22s, min:1.14s, avg: 1.18s, p80:1.22s, p95:1.22s, iters:2, tasks:2, threads:2}                                                                                                                                                                                                                                                              | group by:Column#14, Column#15, funcs:sum(Column#13)->Column#10                                                                                                          | N/A       | N/A  |
|         └─Projection_18          | 2322880.00 | 2322880 | cop[tiflash] |               | tiflash_task:{proc max:791.5ms, min:714.9ms, avg: 753.2ms, p80:791.5ms, p95:791.5ms, iters:37, tasks:2, threads:2}                                                                                                                                                                                                                                                   | cast(test.test.col2, double BINARY)->Column#13, test.test.col1, test.test.col3                                                                                          | N/A       | N/A  |
|           └─TableFullScan_13     | 2322880.00 | 2322880 | cop[tiflash] | table:test    | tiflash_task:{proc max:146.5ms, min:57.9ms, avg: 102.2ms, p80:146.5ms, p95:146.5ms, iters:37, tasks:2, threads:2}                                                                                                                                                                                                                                                    | keep order:false                                                                                                                                                        | N/A       | N/A  |
+----------------------------------+------------+---------+--------------+---------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+------+

**Comment by SeaRise** [comment]
> > I'm not sure if this change compatible with `BatchCorpcessor`, please help confirm this.
> 
> looks ok
> 
> * mpp
> 
> ```
> +------------------------------------------------+------------+---------+--------------+---------------+---------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------+------+
> | id                                             | estRows    | actRows | task         | access object | execution info                                                                                          | operator info                                                                                                                                                           | memory | disk |
> +------------------------------------------------+------------+---------+--------------+---------------+---------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------+------+
> | TableReader_82                                 | 5.17       | 5       | root         |               | time:1.34s, loops:2, cop_task: {num: 2, max: 0s, min: 0s, avg: 0s, p95: 0s, copr_cache_hit_ratio: 0.00} | data:ExchangeSender_81                                                                                                                                                  | N/A    | N/A  |
> | └─ExchangeSender_81                            | 5.17       | 5       | mpp[tiflash] |               | tiflash_task:{time:1.33s, loops:1, threads:20}                                                          | ExchangeType: PassThrough                                                                                                                                               | N/A    | N/A  |
> |   └─Projection_76                              | 5.17       | 5       | mpp[tiflash] |               | tiflash_task:{time:1.32s, loops:1, threads:20}                                                          | Column#8, Column#9                                                                                                                                                      | N/A    | N/A  |
> |     └─HashAgg_74                               | 5.17       | 5       | mpp[tiflash] |               | tiflash_task:{time:1.32s, loops:1, threads:20}                                                          | group by:Column#36, funcs:sum(Column#34)->Column#8, funcs:sum(Column#35)->Column#9                                                                                      | N/A    | N/A  |
> |       └─Projection_84                          | 5.17       | 5       | mpp[tiflash] |               | tiflash_task:{time:1.32s, loops:1, threads:20}                                                          | cast(test.test.col1, decimal(5,0) UNSIGNED BINARY)->Column#34, cast(test.test.col3, decimal(20,0) BINARY)->Column#35, Column#7                                          | N/A    | N/A  |
> |         └─ExchangeReceiver_54                  | 5.17       | 5       | mpp[tiflash] |               | tiflash_task:{time:1.31s, loops:1, threads:20}                                                          |                                                                                                                                                                         | N/A    | N/A  |
> |           └─ExchangeSender_53                  | 5.17       | 5       | mpp[tiflash] |               | tiflash_task:{time:1.31s, loops:1, threads:20}                                                          | ExchangeType: HashPartition, Hash Cols: [name: Column#7, collate: binary]                                                                                               | N/A    | N/A  |
> |             └─Projection_49                    | 5.17       | 5       | mpp[tiflash] |               | tiflash_task:{time:1.31s, loops:1, threads:20}                                                          | Column#7, test.test.col1, test.test.col3                                                                                                                                | N/A    | N/A  |
> |               └─HashAgg_50                     | 5.17       | 5       | mpp[tiflash] |               | tiflash_task:{time:1.31s, loops:1, threads:20}                                                          | group by:test.test.col1, test.test.col3, funcs:sum(Column#19)->Column#7, funcs:firstrow(test.test.col1)->test.test.col1, funcs:firstrow(test.test.col3)->test.test.col3 | N/A    | N/A  |
> |                 └─ExchangeReceiver_52          | 5.17       | 5       | mpp[tiflash] |               | tiflash_task:{time:1.3s, loops:1, threads:20}                                                           |                                                                                                                                                                         | N/A    | N/A  |
> |                   └─ExchangeSender_51          | 5.17       | 5       | mpp[tiflash] |               | tiflash_task:{time:1.31s, loops:1, threads:2}                                                           | ExchangeType: HashPartition, Hash Cols: [name: test.test.col1, collate: binary], [name: test.test.col3, collate: binary]                                                | N/A    | N/A  |
> |                     └─HashAgg_45               | 5.17       | 5       | mpp[tiflash] |               | tiflash_task:{time:1.3s, loops:1, threads:2}                                                            | group by:Column#32, Column#33, funcs:sum(Column#31)->Column#19                                                                                                          | N/A    | N/A  |
> |                       └─Projection_83          | 2322880.00 | 2322880 | mpp[tiflash] |               | tiflash_task:{time:848.4ms, loops:37, threads:2}                                                        | cast(test.test.col2, double BINARY)->Column#31, test.test.col1, test.test.col3                                                                                          | N/A    | N/A  |
> |                         └─TableFullScan_34     | 2322880.00 | 2322880 | mpp[tiflash] | table:test    | tiflash_task:{time:181.4ms, loops:37, threads:2}                                                        | keep order:false                                                                                                                                                        | N/A    | N/A  |
> +------------------------------------------------+------------+---------+--------------+---------------+---------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------+------+
> ```
> 
> * batchCop
> 
> ```
> +----------------------------------+------------+---------+-------------------+---------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+------+
> | id                               | estRows    | actRows | task              | access object | execution info                                                                                                                                                                                                                                                                                                                                                        | operator info                                                                                                                                                           | memory    | disk |
> +----------------------------------+------------+---------+-------------------+---------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+------+
> | HashAgg_9                        | 5.17       | 5       | root              |               | time:1.16s, loops:4, partial_worker:{wall_time:1.158933148s, concurrency:5, task_num:4, tot_wait:5.794392244s, tot_exec:73.671µs, tot_time:5.794488045s, max:1.158902885s, p95:1.158902885s}, final_worker:{wall_time:1.15898255s, concurrency:5, task_num:4, tot_wait:5.794554389s, tot_exec:139.034µs, tot_time:5.794698452s, max:1.158953888s, p95:1.158953888s}   | group by:Column#18, funcs:sum(Column#16)->Column#8, funcs:sum(Column#17)->Column#9                                                                                      | 37.9 KB   | N/A  |
> | └─Projection_19                  | 5.17       | 5       | root              |               | time:1.16s, loops:5, Concurrency:OFF                                                                                                                                                                                                                                                                                                                                  | cast(test.test.col1, decimal(5,0) UNSIGNED BINARY)->Column#16, cast(test.test.col3, decimal(20,0) BINARY)->Column#17, Column#7                                          | 1.11 KB   | N/A  |
> |   └─HashAgg_14                   | 5.17       | 5       | root              |               | time:1.16s, loops:5, partial_worker:{wall_time:1.158761518s, concurrency:5, task_num:1, tot_wait:5.793408843s, tot_exec:28.892µs, tot_time:5.793454968s, max:1.158731855s, p95:1.158731855s}, final_worker:{wall_time:1.158790169s, concurrency:5, task_num:4, tot_wait:5.793568277s, tot_exec:248.29µs, tot_time:5.79382232s, max:1.158774266s, p95:1.158774266s}    | group by:test.test.col1, test.test.col3, funcs:sum(Column#10)->Column#7, funcs:firstrow(test.test.col1)->test.test.col1, funcs:firstrow(test.test.col3)->test.test.col3 | 17.3 KB   | N/A  |
> |     └─TableReader_15             | 5.17       | 5       | root              |               | time:1.16s, loops:2, cop_task: {num: 2, max: 0s, min: 0s, avg: 0s, p95: 0s, copr_cache_hit_ratio: 0.00, distsql_concurrency: 15}                                                                                                                                                                                                                                      | data:HashAgg_10                                                                                                                                                         | 152 Bytes | N/A  |
> |       └─HashAgg_10               | 5.17       | 5       | batchCop[tiflash] |               | tiflash_task:{time:1.15s, loops:1, threads:2}                                                                                                                                                                                                                                                                                                                         | group by:Column#14, Column#15, funcs:sum(Column#13)->Column#10                                                                                                          | N/A       | N/A  |
> |         └─Projection_18          | 2322880.00 | 2322880 | batchCop[tiflash] |               | tiflash_task:{time:743.9ms, loops:37, threads:2}                                                                                                                                                                                                                                                                                                                      | cast(test.test.col2, double BINARY)->Column#13, test.test.col1, test.test.col3                                                                                          | N/A       | N/A  |
> |           └─TableFullScan_13     | 2322880.00 | 2322880 | batchCop[tiflash] | table:test    | tiflash_task:{time:77.9ms, loops:37, threads:2}                                                                                                                                                                                                                                                                                                                       | keep order:false                                                                                                                                                        | N/A       | N/A  |
> +----------------------------------+------------+---------+-------------------+---------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+------+
> ```
> 
> * cop
> 
> ```
> +----------------------------------+------------+---------+--------------+---------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+------+
> | id                               | estRows    | actRows | task         | access object | execution info                                                                                                                                                                                                                                                                                                                                                       | operator info                                                                                                                                                           | memory    | disk |
> +----------------------------------+------------+---------+--------------+---------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+------+
> | HashAgg_9                        | 5.17       | 5       | root         |               | time:1.23s, loops:4, partial_worker:{wall_time:1.225490746s, concurrency:5, task_num:4, tot_wait:6.127030474s, tot_exec:31.683µs, tot_time:6.127071437s, max:1.225452897s, p95:1.225452897s}, final_worker:{wall_time:0s, concurrency:5, task_num:4, tot_wait:6.12737545s, tot_exec:80.303µs, tot_time:6.127458778s, max:1.225499416s, p95:1.225499416s}             | group by:Column#18, funcs:sum(Column#16)->Column#8, funcs:sum(Column#17)->Column#9                                                                                      | 37.9 KB   | N/A  |
> | └─Projection_19                  | 5.17       | 5       | root         |               | time:1.23s, loops:5, Concurrency:OFF                                                                                                                                                                                                                                                                                                                                 | cast(test.test.col1, decimal(5,0) UNSIGNED BINARY)->Column#16, cast(test.test.col3, decimal(20,0) BINARY)->Column#17, Column#7                                          | 1.11 KB   | N/A  |
> |   └─HashAgg_14                   | 5.17       | 5       | root         |               | time:1.23s, loops:5, partial_worker:{wall_time:1.225280629s, concurrency:5, task_num:1, tot_wait:6.12612229s, tot_exec:37.253µs, tot_time:6.126172144s, max:1.225264616s, p95:1.225264616s}, final_worker:{wall_time:1.225299391s, concurrency:5, task_num:4, tot_wait:6.126260169s, tot_exec:143.743µs, tot_time:6.12640806s, max:1.225288809s, p95:1.225288809s}   | group by:test.test.col1, test.test.col3, funcs:sum(Column#10)->Column#7, funcs:firstrow(test.test.col1)->test.test.col1, funcs:firstrow(test.test.col3)->test.test.col3 | 17.3 KB   | N/A  |
> |     └─TableReader_15             | 5.17       | 10      | root         |               | time:1.23s, loops:2, cop_task: {num: 2, max: 1.23s, min: 1.15s, avg: 1.19s, p95: 1.23s, rpc_num: 2, rpc_time: 2.38s, copr_cache_hit_ratio: 0.00, distsql_concurrency: 15}                                                                                                                                                                                            | data:HashAgg_10                                                                                                                                                         | 638 Bytes | N/A  |
> |       └─HashAgg_10               | 5.17       | 10      | cop[tiflash] |               | tiflash_task:{proc max:1.22s, min:1.14s, avg: 1.18s, p80:1.22s, p95:1.22s, iters:2, tasks:2, threads:2}                                                                                                                                                                                                                                                              | group by:Column#14, Column#15, funcs:sum(Column#13)->Column#10                                                                                                          | N/A       | N/A  |
> |         └─Projection_18          | 2322880.00 | 2322880 | cop[tiflash] |               | tiflash_task:{proc max:791.5ms, min:714.9ms, avg: 753.2ms, p80:791.5ms, p95:791.5ms, iters:37, tasks:2, threads:2}                                                                                                                                                                                                                                                   | cast(test.test.col2, double BINARY)->Column#13, test.test.col1, test.test.col3                                                                                          | N/A       | N/A  |
> |           └─TableFullScan_13     | 2322880.00 | 2322880 | cop[tiflash] | table:test    | tiflash_task:{proc max:146.5ms, min:57.9ms, avg: 102.2ms, p80:146.5ms, p95:146.5ms, iters:37, tasks:2, threads:2}                                                                                                                                                                                                                                                    | keep order:false                                                                                                                                                        | N/A       | N/A  |
> +----------------------------------+------------+---------+--------------+---------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+------+
> ```

- mpp
```
+-------------------------+------------+---------+--------------+---------------+-------------------------------------------------------------------------------------------------------------+---------------------------+--------+------+
| id                      | estRows    | actRows | task         | access object | execution info                                                                                              | operator info             | memory | disk |
+-------------------------+------------+---------+--------------+---------------+-------------------------------------------------------------------------------------------------------------+---------------------------+--------+------+
| TableReader_9           | 2322880.00 | 2322880 | root         |               | time:1.33s, loops:2270, cop_task: {num: 38, max: 0s, min: 0s, avg: 0s, p95: 0s, copr_cache_hit_ratio: 0.00} | data:ExchangeSender_8     | N/A    | N/A  |
| └─ExchangeSender_8      | 2322880.00 | 2322880 | mpp[tiflash] |               | tiflash_task:{time:1.31s, loops:37, threads:2}                                                              | ExchangeType: PassThrough | N/A    | N/A  |
|   └─TableFullScan_7     | 2322880.00 | 2322880 | mpp[tiflash] | table:test    | tiflash_task:{time:65.1ms, loops:37, threads:2}                                                             | keep order:false          | N/A    | N/A  |
+-------------------------+------------+---------+--------------+---------------+-------------------------------------------------------------------------------------------------------------+---------------------------+--------+------+
```
- batchCop
```
+-----------------------+------------+---------+-------------------+---------------+--------------------------------------------------------------------------------------------------------------------------------------+----------------------+---------+------+
| id                    | estRows    | actRows | task              | access object | execution info                                                                                                                       | operator info        | memory  | disk |
+-----------------------+------------+---------+-------------------+---------------+--------------------------------------------------------------------------------------------------------------------------------------+----------------------+---------+------+
| TableReader_5         | 2322880.00 | 2322880 | root              |               | time:3.03s, loops:2270, cop_task: {num: 38, max: 0s, min: 0s, avg: 0s, p95: 0s, copr_cache_hit_ratio: 0.00, distsql_concurrency: 15} | data:TableFullScan_4 | 2.32 MB | N/A  |
| └─TableFullScan_4     | 2322880.00 | 2322880 | batchCop[tiflash] | table:test    | tiflash_task:{time:230.3ms, loops:37, threads:2}                                                                                     | keep order:false     | N/A     | N/A  |
+-----------------------+------------+---------+-------------------+---------------+--------------------------------------------------------------------------------------------------------------------------------------+----------------------+---------+------+
```
- cop
```
+-----------------------+------------+---------+--------------+---------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------------------+---------+------+
| id                    | estRows    | actRows | task         | access object | execution info                                                                                                                                                               | operator info        | memory  | disk |
+-----------------------+------------+---------+--------------+---------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------------------+---------+------+
| TableReader_5         | 2322880.00 | 2322880 | root         |               | time:1.96s, loops:2271, cop_task: {num: 2, max: 1.96s, min: 1.78s, avg: 1.87s, p95: 1.96s, rpc_num: 2, rpc_time: 3.73s, copr_cache_hit_ratio: 0.00, distsql_concurrency: 15} | data:TableFullScan_4 | 82.1 MB | N/A  |
| └─TableFullScan_4     | 2322880.00 | 2322880 | cop[tiflash] | table:test    | tiflash_task:{proc max:200.5ms, min:69ms, avg: 134.7ms, p80:200.5ms, p95:200.5ms, iters:37, tasks:2, threads:2}                                                              | keep order:false     | N/A     | N/A  |
+-----------------------+------------+---------+--------------+---------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------------------+---------+------+
```

**Review by windtalker — APPROVED** [review]
LGTM

**Review by ywqzzy — APPROVED** [review]

**Pull request merged**